### ISSUE TITLE
Support for real audio bibles

### ISSUE THREAD

**Issue by tse-gratis** [AndBible/and-bible#1909]
The people we most often give bible apps cannot read in their own language. So i choose the faith comes by hearing bible apps for them. They are simple and the audio quality is good

For those apps, there is no way to download the audio in one go!!!! 100% of those people have not had wifi or data, so a percentage of every meeting is spent with us clicking on as many chapters as possible, and waiting for that chapter's audio to download

That is painful. They get at most 30mins of audio a week that way, and the interface does not tell them which passages have audio and which dont

From the And Bible language list, it looks like the SWORD project has imported the faith comes by hearing bible resources... I'm guessing, but would it be possible to also import their audio resources? Or in someway get one-click download of their audio?

That would make And Bible excellent for this usecase

**Comment by tuomas2** [comment]
I don't know if they would allow us to use their audio. 

Anyway, it would be big task to do this.

**Comment by agrogers** [comment]
Wow, that's amazing that you can get the NT for ESV for free. Dramatised as
well. I might replace my copy with that.

Tuomas, how about functionality that allowed users to point AndBible to a
folder which they have already created and downloaded the audio files. Then
there are no licensing issues for us. We just look in that folder and play
what is there. If it could be left as a zip file and we just look inside
the zip it would be even better.

___________________________
*www.TenOutOfTen.org* <https://www.TenOutOfTen.org>
***@***.***
(+61) 0477 174 193 (Australia)
(+95) 09 250018669 (Myanmar)

**Comment by tuomas2** [comment]
feel free to implement 😁

**Comment by agrogers** [comment]
I know you like the Android voice :) I just listened to an hour of my ESV
from Christian Audio... it's hard to go back to a computer voice after
listening to a human.
Implement one day perhaps ;-)

___________________________
*www.TenOutOfTen.org* <https://www.TenOutOfTen.org>
***@***.***
(+61) 0477 174 193 (Australia)
(+95) 09 250018669 (Myanmar)

**Comment by tuomas2** [comment]
> I know you like the Android voice :) I just listened to an hour of my ESV from Christian Audio... it's hard to go back to a computer voice after listening to a human. Implement one day perhaps ;-)

Try another open source app called "Voice" (https://f-droid.org/en/packages/de.ph1b.audiobook/). It has everything you need for listening to audiobooks.

Those mp3 files do not have any metadata that helps us to know where each individual verse starts, making it impossible to implement this feature properly. To obtain that data we probably need to negotiate licenses. And even if we get licenses, it will still be huge task. You need to design it properly and implement it seamlessly to the current TTS-based solution. It's nice to have if someone does it properly but not very likely to happen, I would say.

Not that I like TTS voice, but it's generally "good enough" for getting the real content. And there are not (good) finnish audio bibles anyway... (okay, there is one for 1992 translation that I am not interested to use).

**Comment by agrogers** [comment]
I use Smart Audio player. It's great too.

I have heard the request to have a real human read the text a few times
now. And I am sympathetic to that. I would be happy just to be able to
launch my audio player to play the correct chapter file. Most reading plans
are chapter based arent they. I shall ponder it... for a long time i
suspect.

___________________________
*www.TenOutOfTen.org* <https://www.TenOutOfTen.org>
***@***.***
(+61) 0477 174 193 (Australia)
(+95) 09 250018669 (Myanmar)

**Comment by tse-gratis** [comment]
It seems that streaming does not require a license, but download does for some resources (possibly per chapter download is okay). I am asking them

Per verse audio seems possible

Their api:
https://www.faithcomesbyhearing.com/audio-bible-resources/bible-brain

Also here:
https://github.com/faithcomesbyhearing/dbp

TTS has advantages though -- size of download, offline playback. I'd first like to check for a good multi-lingual TTS

**Comment by tuomas2** [comment]
Using API goes to some extent against having AB fully offline

**Comment by eigenlambda** [comment]
(1) feed alexander scourby to speech to text engine
(2) match up text with kjv
(3) determine verse markers on word boundaries
(4) automatically have alexander scourby read the kjv starting anywhere
but, since alexander scourby has a copyright on his reading, we need to provide the algorithms and allow the user to supply the mp3s for each Bible book and have some way of dealing with when scourby says "the Holy Bible, the King James Version, the New Testament, the book of Matthew, chapter 1"

**Comment by Noitarud** [comment]
When you choose listen, you could be given a choice of resources (define these already in settings - their location, folder layout and playlist/chapter file organisation if necessary), besides the TTS.
The resource folders could be just an audio file (NIV OT is 49H42M, NT is 16H13M) or books or chapters.
When there are chapter files or individual chapters defined (in file imetadata or as a cue file) infer verse by percentage.

Using FFMPEG I have added some into the metadata of mine (Gen 1,11,25,37, Exo 1,19,25,3235…) so "next" jumps are not so large. The user could define the missing chapter offsets themselves, at least on their favourite places. Maybe a git topic for sharing timings - Ideally you should share free speakers of free Bibles else everyone may be admitting to breaking law (NIV is owned by the Murdoch conglomerate and they have obstructed missionaries - source may have been Gail Riplinger). I have a copy of Brad Moyers reading the ASV new testament in chapters, it is free (cannot verify if still downloadable).

Some people may want the speed slower (or faster) so ensure fractional adjustment is available.

If you have a free Bible, a steady voice and ¾ day you could record the NT or just the gospels, you don't need an actor's voice (just make sure you get Mat6:12 right, it will sound better if the speeches were recorded separate (tedious), and I am positive the KJV is supposed to be read with a Cockney accent). Record yourself reading to your children Deu 6:4-9,11:19, you can use dual microphones to remove sounds that are to your left and right (ensure noone is ahead of you).

**Comment by timbze** [comment]
Soon it will likely be possible to use offline AI voices. https://github.com/KittenML/KittenTTS

**Comment by agrogers** [comment]
> Using API goes to some extent against having AB fully offline

I don't think this is still the case, yes? AI prominence brings online to the forefront.

Ensuring key functionality remains offline is what matters I think.

**Comment by meichthys** [comment]
With mobile devices getting so powerful and small llms improving, It may be possible eventually to run the AI processon on-device. It'll never be as powerful as cloud models, but for simple lookups like cross references, or better TTS, it may be 'good enough'.

**Comment by tuomas2** [comment]
AI-tts "audio bible" probably better than real one. It would be also possible to make integration to OpenAI:s whisper api (then OpenAI:s API key would enable it). 

I'm not really into making AndBible online app (meaning, should set up server for online features too). Though that is not completely out of question, as long as features stay completely optional (and opt-in).
